tests/test_validation.py

```diff
@@ -37,7 +37,10 @@ class Author(InnerDoc):
     email = Text(required=True)
 
     def clean(self):
-        print(self, type(self), self.name)
+        if not self.name:
+            raise ValidationException("name is missing")
+        if not self.email:
+            raise ValidationException("email is missing")
         if self.name.lower() not in self.email:
```

Comment:
I like making it clear when branches are exclusive:

```suggestion
        if not self.email:
            raise ValidationException("email is missing")
        elif self.name.lower() not in self.email:
```

(Also applies to the next if)